lib/ansible/modules/cloud/amazon/iam_user_info.py

```diff
@@ -0,0 +1,184 @@
+            returned: if password was used at least once
+            type: str
+            sample: "2016-05-25T13:39:11+00:00"
+        path:
+            description: the path to user
+            returned: if user exists
+            type: str
+            sample: "/dev/"
+        user_id:
+            description: the unique user id
+            returned: if user exists
+            type: str
+            sample: "AIDUIOOCQKTUGI6QJLGH2"
+        user_name:
+            description: the user name
+            returned: if user exists
+            type: str
+            sample: "test_user"
+'''
+
+from ansible.module_utils.aws.core import AnsibleAWSModule
+from ansible.module_utils.ec2 import boto3_conn, camel_dict_to_snake_dict, ec2_argument_spec, get_aws_connection_info
+
+try:
+    import botocore
+    from botocore.exceptions import ClientError, ParamValidationError
+except ImportError:
+    pass  # caught by AnsibleAWSModule
+
+
+def list_iam_users(connection, module):
+
+    name = module.params.get('name')
+    group = module.params.get('group')
+    path = module.params.get('path')
+
+    params = dict(MaxItems=1000)
+    iam_users = []
+
+    if not group and not path:
+        if name:
+            params['UserName'] = name
+        try:
+            iam_users.append(connection.get_user(**params)['User'])
+        except (ClientError, ParamValidationError) as e:
+            module.fail_json_aws(e, msg="Couldn't get IAM user info for user %s" % name)
+
+    if group:
+        params['GroupName'] = group
+        try:
+            iam_users = connection.get_group(**params)['Users']
+        except (ClientError, ParamValidationError) as e:
+            module.fail_json_aws(e, msg="Couldn't get IAM user info for group %s" % group)
+        if name:
+            iam_users = [user for user in iam_users if user['UserName'] == name]
+
+    if path and not group:
+        params['PathPrefix'] = path
+        try:
+            iam_users = connection.list_users(**params)['Users']
+        except (ClientError, ParamValidationError) as e:
+            module.fail_json_aws(e, msg="Couldn't get IAM user info for path %s" % path)
+        if name:
+            iam_users = [user for user in iam_users if user['UserName'] == name]
+
+    module.exit_json(iam_users=[camel_dict_to_snake_dict(user) for user in iam_users])
+
+
+def main():
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(
+        dict(
+            name=dict(),
+            group=dict(),
+            path=dict(default='/')
+        )
+    )
+
+    module = AnsibleAWSModule(
+        argument_spec=argument_spec,
+        mutually_exclusive=[
+            ['group', 'path']
+        ],
+        supports_check_mode=True
+    )
+
+    region, ec2_url, aws_connect_params = get_aws_connection_info(module, boto3=True)
+
+    connection = boto3_conn(module, conn_type='client', resource='iam', region=region, endpoint=ec2_url, **aws_connect_params)
```

Comment:
AnsibleAWSModule provides a nice helper
```suggestion
    connection = module.client('iam')
```

Reply:
https://github.com/ansible/ansible/pull/23382/commits/0fd12989d28ed8304e6cdaaa1b9437d32dc08867 changes this.